From 8db77add1a794bdee8eef0a351e40bf1cdf6dfa9 Mon Sep 17 00:00:00 2001
From: Robert Marko <robimarko@gmail.com>
Date: Sun, 22 Jan 2023 22:09:51 +0100
Subject: [PATCH 4/5] cryptoapi: v2.0: aead: add downstream
 crypto_tfm_alg_flags

crypto_tfm_alg_flags newer made it upstream, but as a temporary stopgap
until a better solution is figured out lets add it.

Signed-off-by: Robert Marko <robimarko@gmail.com>
---
 cryptoapi/v2.0/nss_cryptoapi_aead.c | 5 +++++
 1 file changed, 5 insertions(+)

--- a/cryptoapi/v2.0/nss_cryptoapi_aead.c
+++ b/cryptoapi/v2.0/nss_cryptoapi_aead.c
@@ -61,6 +61,11 @@
 #include <nss_cryptoapi.h>
 #include "nss_cryptoapi_private.h"
 
+static inline u32 crypto_tfm_alg_flags(struct crypto_tfm *tfm)
+{
+	return tfm->__crt_alg->cra_flags & ~CRYPTO_ALG_TYPE_MASK;
+}
+
 /*
  * nss_cryptoapi_aead_ctx2session()
  *	Cryptoapi function to get the session ID for an AEAD
